dist/aframe-altspace-component.js

```diff
@@ -465,86 +467,744 @@
+					if (tagName === "a-animation") {
+						stoppedAnimations.push(children[i]);
+						children[i].stop();
+					}
+				}
+			});
+			this.el.addEventListener('ownershipgained', function () {
+				for (var i = 0; i < stoppedAnimations.length; i++) {
+					var animation = stoppedAnimations[i];
+					animation.start();
+				}
+				stoppedAnimations = [];
+			});
+
+			positionRef.on('value', function (snapshot) {
+				if (sync.isMine) return;
+
+				var value = snapshot.val();
+				if (!value) return;
+
+				component.el.setAttribute('position', {x: value.x, y: value.y, z: value.z});
+			});
+
+			rotationRef.on('value', function (snapshot) {
+				if (sync.isMine) return;
+
+				var value = snapshot.val();
+				if (!value) return;
+
+				component.el.setAttribute('rotation', {x: value.x, y: value.y, z: value.z});
+			});
+
+			scaleRef.on('value', function (snapshot) {
+				if (sync.isMine) return;
+
+				var value = snapshot.val();
+				if (!value) return;
+
+				component.el.setAttribute('scale', {x: value.x, y: value.y, z: value.z});
+			});
+
+			function onComponentChanged(event){
+				if (!sync.isMine) return;
+
+				var name = event.detail.name;
+				var oldData = event.detail.oldData;
+				var newData = event.detail.newData;
+
+				if (name === 'position') {
+					positionRef.set({
+						x: newData.x,
+						y: newData.y,
+						z: newData.z
+					});
+				} else if (name === 'rotation') {
+					rotationRef.set({
+						x: newData.x,
+						y: newData.y,
+						z: newData.z
+					});
+				} else if (name === 'scale') {
+					scaleRef.set({
+						x: newData.x,
+						y: newData.y,
+						z: newData.z
+					});
+				} else {
+					return;
+				}
+
+			}
+
+			//from underscore.js
+			function throttle(func, wait, options) {
+				var timeout, context, args, result;
+				var previous = 0;
+				if (!options) options = {};
+
+				var later = function() {
+				  previous = options.leading === false ? 0 : Date.now();
+				  timeout = null;
+				  result = func.apply(context, args);
+				  if (!timeout) context = args = null;
+				};
+
+				var throttled = function() {
+				  var now = Date.now();
+				  if (!previous && options.leading === false) previous = now;
+				  var remaining = wait - (now - previous);
+				  context = this;
+				  args = arguments;
+				  if (remaining <= 0 || remaining > wait) {
+					if (timeout) {
+					  clearTimeout(timeout);
+					  timeout = null;
+					}
+					previous = now;
+					result = func.apply(context, args);
+					if (!timeout) context = args = null;
+				  } else if (!timeout && options.trailing !== false) {
+					timeout = setTimeout(later, remaining);
+				  }
+				  return result;
+				};
+
+				throttled.cancel = function() {
+				  clearTimeout(timeout);
+				  previous = 0;
+				  timeout = context = args = null;
+				};
+
+				return throttled;
+			  };
+
+
+			this.el.addEventListener('componentchanged', throttle(onComponentChanged, this.updateRate));
```

Comment:
Well, that was an N^2 thing. But, ok. I guess it's fine if it doesn't visually affect animations too much. We'd probably want to add updateRate to the component schema though.